### ISSUE TITLE
`ssh git@data.neuro.polymtl.ca pr`

### ISSUE THREAD

**Issue by kousu** [neuropoly/data-management#86]
Write a gitolite extension to implement pull requests for the private data server.

We can't use `git send-email` because we don't have a mailing list, and anyway the bulk of our data are large binaries in `.git/annex` which don't email well :P. So right now "PRs" are someone pinging me or @alexfoias on Slack, or maybe opening an issue in this repo and tagging us there.

**Comment by kousu** [comment]
Related work:

* radicle: https://docs.radicle.xyz/docs/understanding-radicle/faq/#how-will-issues-and-prs-work
    * huh, funny, i thought they had PRs working? maybe I saw an alpha version of PRs complete with comments, but maybe they've backed out of them
    * planning to have a CLI-based PR workflow
* Gogs: https://gogs.io/ - clone of github
    * web-based PR workflow
* multiverse: https://www.multiverse-vcs.com/docs/
    * no PRs yet? just merges?

**Comment by jcohenadad** [comment]
Wow, the examples listed in https://github.com/neuropoly/data-management/issues/86#issuecomment-854941206 are really cool! if we could have such interface that would be wonderful. Otherwise we can just use our current private neuropoly/data_management hosted on github.

**Comment by kousu** [comment]
Here's a sketch of the UI:

```
ssh git@data.neuro.polymtl.ca pr open datasets/uk-biobank ng/add-some-stuff
ssh git@data.neuro.polymtl.ca pr ls datasets/uk-biobank # shows open PRs
ssh git@data.neuro.polymtl.ca pr ls --open datasets/uk-biobank # same as previous
ssh git@data.neuro.polymtl.ca pr ls --merged datasets/uk-biobank
ssh git@data.neuro.polymtl.ca pr ls --closed datasets/uk-biobank
ssh git@data.neuro.polymtl.ca pr comment datasets/uk-biobank "I think you should change ..."
ssh git@data.neuro.polymtl.ca pr comments datasets/uk-biobank ng/add-some-stuff
ssh git@data.neuro.polymtl.ca pr approve datasets/uk-biobank ng/add-some-stuff ["optional message"]
ssh git@data.neuro.polymtl.ca pr close datasets/uk-biobank ng/add-some-stuff ["optional message"]
ssh git@data.neuro.polymtl.ca pr open datasets/uk-biobank ng/add-some-stuff ["optional message"] #reopens
```

With my [ssh tip](https://github.com/neuropoly/neuropoly-docs/pull/8) you can shorten this to e.g.

```
ssh data pr ls
```
